### PR TITLE
Fwa 004 handle data from json

### PR DESCRIPTION
**Pull request by heupotato**

### CODE REVIEW

File: Flutter_Weather/assets/jsons/mock_json_data.json [hunk 1]
```diff
@@ -0,0 +1,902 @@
+﻿{
```
Comment (lngyeen) [inline Flutter_Weather/assets/jsons/mock_json_data.json]: File name should describe file contents. (Should be 'mock_weather.json')

File: Flutter_Weather/jsons/mockjsondata.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (lngyeen) [inline Flutter_Weather/jsons/mockjsondata.json]: This is a json data model file,  not a mock data file. So the name should be 'weather.json'

File: Flutter_Weather/lib/models/weather_data.dart [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+  @override
+  String toString() {
+    super.toString();
+    return("$timepoint $cloudcover");
+  }
+}
+
+@immutable
+class Wind10m {
```
Comment (lngyeen) [inline Flutter_Weather/lib/models/weather_data.dart]: Should create json file 'wind10m.json' put it in jsons/ folder. So json_to_model will generate separated files for each model. Easy to maintain code.

File: Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    final data = await jsonDecode(response);
+    Mockjsondata mockdata = Mockjsondata.fromJson(data);
+    return mockdata;
+  }
+}
+
+extension ProcessMockJsonData on Mockjsondata{
+  //int getInitTime() => int.parse(init.substring(init.length - 2, init.length));
+
+  DateTime getInitDate() => DateFormatter.yearMonthDayHour(init);//DateFormatter.yearMonthDay(init.substring(0, init.length -2));
```
Comment (lngyeen) [inline Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart]: Should be get property, not function: 'DateTime get initDate => /...'

File: Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+  DateTime getTrueDate(int timePoint) => this.getInitDate().add(Duration(hours: timePoint));
```
Comment (lngyeen) [inline Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart]: Same above, should be get property

File: Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+  List<DateTime> getTrueDates(List<WeatherData> dataList)
```
Comment (lngyeen) [inline Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart:26]: same above

File: Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+   => dataList.map((data) => this.getTrueDate(data.timepoint)).toList();
+
+  List<String> getWeekDays(List<WeatherData> dataList)
```
Comment (lngyeen) [inline Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart]: same above

File: Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+  List<WeatherData> getWeekData(){
+    final indexList = this.getDateIndex();
+    List<WeatherData> weekData = [];
+    for (int i = 0; i < 8; i++)
+      weekData.add(dataseries[indexList[i]]);
+    return weekData;
+  }
+
+  List<WeatherData> getDayData(){
```
Comment (lngyeen) [inline Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart:53]: Day data some time has only 1 or 2 time points, not always 8. 
For example: Weather data was created at 6PM UTC time, so today, we only have 2 weather time points [3, 6]. (Real time of each weather item will be:  WeatherData's initTime + item's time point).
 You should create a class 'DayWeather' to handle this.

File: Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+  List<int> getDateIndex(){
+    List<int> indexes = [];
+    for (int i = 0; i < dataseries.length; i++){
+      if (i%8==0) indexes.add(i);
+    }
+    return indexes;
+  }
+
+  List<WeatherData> getWeekData(){
```
Comment (lngyeen) [inline Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart:39]: Should create class 'WeekWeather' contains List of DayWeather to handle data for a week.

**Review by lngyeen — CHANGES_REQUESTED** [review]

File: Flutter_Weather/lib/services/date_formatter.dart [hunk 10]
```diff
@@ -18,6 +18,15 @@ class DateFormatter {
   static String weekDay(DateTime dateTime) =>
       DateFormat('EEEE').format(dateTime);
 
-  static DateTime yearMonthDayHour(String initDate) =>
-      DateFormat('yyyy MM dd hh').parse(initDate, true);
+  static DateTime yearMonthDayHour(String initDate){
```
Comment (lngyeen) [inline Flutter_Weather/lib/services/date_formatter.dart]: Should move to Weather class extension, This special value handling occurs only in that class, it is specific to that class.

**Review by lngyeen — CHANGES_REQUESTED** [review]

**Comment by lngyeen** [comment]
Resolve conflicts.

**Comment by heupotato** [comment]
Conflicts resolved

**Pull request merged**